### PR TITLE
ui: Remove bidirectional dependencies in UiParts #448

### LINKED ISSUE #448: ui: Remove bidirectional dependencies in UiParts
Most `UiParts` (e.g. `CommandBox`, `PersonListPanel` etc.) take a "placeholder" argument in their constructor. The parent of these UiParts (e.g. the `MainWindow`) will then pass in a pane through this argument, and the `UiPart` will add itself to it. This creates a bi-directional dependency between the parent and child (i.e. UiParts need a parent placeholder to exist).

The current situation works, but its not very clean, since it requires an additional argument in constructors.

Very coincidentally, all `UiParts` have a [`getRoot()`](https://github.com/se-edu/addressbook-level4/blob/6af393367c998910784881146ffa049725bc31bd/src/main/java/seedu/address/ui/UiPart.java#L51) getter which returns the root of their scene graph. Using this method, parents can directly add the child UiPart's root to the placeholder pane, without needing to go through the child UiPart's constructor. This will eliminate the bi-directional dependency and reduce the number of constructor arguments of UiParts.

### LINKED ISSUE #517: CommandBox: Remove SplitPane#setResizableWithParent(…) method call
Both commandBoxPlaceholder (in MainWindow.fxml) and CommandBox (in CommandBox.fxml) do not use SplitPane.

As such, this method call does nothing.

Let’s remove it.

### PR DESCRIPTION
**Pull request by yamgent**
Fixes #448. Also fixes #517.

Proposed merge commit message:
```
The components accept a placeholder in their parameter in order for them
to add themselves into the placeholder. Therefore the placeholder acts
like a parent to the component.

The components themselves should not know about their parent placeholder
as this creates a bi-directional dependency coupling.

Let's remove the placeholder parameter in all component's constructor, and
let MainWindow add the component to the placeholder instead.

The components no longer fills up the entire space of the placeholder as
it now contains the root node of the component as well. Let's also
modify the .fxml to allow the component to fill up the entire space.
```

### CODE REVIEW

**Comment by CanIHasReview** [comment]
# v1

@yamgent submitted v1 for review.

* [v1 1/5] [CommandBox: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/0a1c8a0f591f2bf025c3ffa52aa208db525469d9)
* [v1 2/5] [StatusBarFooter: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/24fbde293c0a1bb6cd12e7be0139b292009bace3)
* [v1 3/5] [ResultDisplay: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/8dc5e5285c02685aeb46a65821bd6a17d3820cd2)
* [v1 4/5] [PersonListPanel: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/ffb0ad8ef96043610b6e907a2dcaff93c4327e64)
* [v1 5/5] [BrowserPanel: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/0f5010ac9b515f6c06fb77d34e169a72b5a06338)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/5cc85fcf11b40d1660d0b6be20a3ef90bc71c1c1...0f5010ac9b515f6c06fb77d34e169a72b5a06338))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/497/1/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 1]
```diff
@@ -116,18 +117,22 @@ private void setAccelerator(MenuItem menuItem, KeyCombination keyCombination) {
     }
 
     void fillInnerParts() {
```
Comment (Zhiyuan-Amos) [inline src/main/java/seedu/address/ui/MainWindow.java]: can have the next commit to clean up this part? haha looks so untidy

**Comment by CanIHasReview** [comment]
# v2

@yamgent submitted v2 for review.

* [v2 1/6] [CommandBox: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/0a1c8a0f591f2bf025c3ffa52aa208db525469d9)
* [v2 2/6] [StatusBarFooter: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/24fbde293c0a1bb6cd12e7be0139b292009bace3)
* [v2 3/6] [ResultDisplay: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/8dc5e5285c02685aeb46a65821bd6a17d3820cd2)
* [v2 4/6] [PersonListPanel: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/ffb0ad8ef96043610b6e907a2dcaff93c4327e64)
* [v2 5/6] [BrowserPanel: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/0f5010ac9b515f6c06fb77d34e169a72b5a06338)
* [v2 6/6] [MainWindow: Refactor out component creation logic](https://github.com/se-edu/addressbook-level4/pull/497/commits/044e6d82435e3ee9e1d25602b1f50922189d0779)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/5cc85fcf11b40d1660d0b6be20a3ef90bc71c1c1...044e6d82435e3ee9e1d25602b1f50922189d0779)) ([:chart_with_upwards_trend: Interdiff between v1 and v2](https://github.com/se-edu/addressbook-level4/compare/1ab1517bf17db48402a4acb17f942d7003400ecb~1...1ab1517bf17db48402a4acb17f942d7003400ecb))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/497/2/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 2]
```diff
@@ -117,22 +117,60 @@ private void setAccelerator(MenuItem menuItem, KeyCombination keyCombination) {
...
+        setupBrowserPanel();
+        setupPersonListPanel();
+        setupResultDisplay();
+        setupStatusBarFooter();
+        setupCommandBox();
+    }
+
+    /**
+     * Create a new browser panel and place it inside its placeholder.
```
Comment (Zhiyuan-Amos) [inline src/main/java/seedu/address/ui/MainWindow.java]: <pre>
Create<b>s</b> a {@code BrowserPanel} and place<b>s</b> it inside its placeholder.
</pre>

Same for the rest.

**Comment by CanIHasReview** [comment]
# v3

@yamgent submitted v3 for review.

* [v3 1/6] [CommandBox: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/0a1c8a0f591f2bf025c3ffa52aa208db525469d9)
* [v3 2/6] [StatusBarFooter: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/24fbde293c0a1bb6cd12e7be0139b292009bace3)
* [v3 3/6] [ResultDisplay: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/8dc5e5285c02685aeb46a65821bd6a17d3820cd2)
* [v3 4/6] [PersonListPanel: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/ffb0ad8ef96043610b6e907a2dcaff93c4327e64)
* [v3 5/6] [BrowserPanel: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/0f5010ac9b515f6c06fb77d34e169a72b5a06338)
* [v3 6/6] [MainWindow: Refactor out component creation logic](https://github.com/se-edu/addressbook-level4/pull/497/commits/260890baee8a83ea85265e8fabee9b392d8aea6b)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/5cc85fcf11b40d1660d0b6be20a3ef90bc71c1c1...260890baee8a83ea85265e8fabee9b392d8aea6b)) ([:chart_with_upwards_trend: Interdiff between v2 and v3](https://github.com/se-edu/addressbook-level4/compare/76f2c66bed907bf20d06d2ad8e79b4b3ed80fd8a~1...76f2c66bed907bf20d06d2ad8e79b4b3ed80fd8a))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/497/3/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

File: src/main/java/seedu/address/ui/MainWindow.java (hunk 2 above)
Comment (Zhiyuan-Amos) [inline src/main/java/seedu/address/ui/MainWindow.java]: @yamgent sorry that I didn't bold the `{@code BrowserPanel}`. :P You can decide whether to edit it; I'll just approve.

**Comment by CanIHasReview** [comment]
# v4

@yamgent submitted v4 for review.

* [v4 1/6] [CommandBox: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/0a1c8a0f591f2bf025c3ffa52aa208db525469d9)
* [v4 2/6] [StatusBarFooter: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/24fbde293c0a1bb6cd12e7be0139b292009bace3)
* [v4 3/6] [ResultDisplay: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/8dc5e5285c02685aeb46a65821bd6a17d3820cd2)
* [v4 4/6] [PersonListPanel: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/ffb0ad8ef96043610b6e907a2dcaff93c4327e64)
* [v4 5/6] [BrowserPanel: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/0f5010ac9b515f6c06fb77d34e169a72b5a06338)
* [v4 6/6] [MainWindow: Refactor out component creation logic](https://github.com/se-edu/addressbook-level4/pull/497/commits/1f71113b807c822b96b01f3ac2ceb6e119f8a6ce)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/5cc85fcf11b40d1660d0b6be20a3ef90bc71c1c1...1f71113b807c822b96b01f3ac2ceb6e119f8a6ce)) ([:chart_with_upwards_trend: Interdiff between v3 and v4](https://github.com/se-edu/addressbook-level4/compare/0bfb8d346c729710d8fe3c9e4f622b4e4272f7fd~1...0bfb8d346c729710d8fe3c9e4f622b4e4272f7fd))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/497/4/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

**Comment by yamgent** [comment]
@limmlingg Reminder for review thanks.

**Comment by CanIHasReview** [comment]
# v5

@yamgent submitted v5 for review.

* [v5 1/6] [CommandBox: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/8f76767089be1f41cffb17a9274314a16ff0ead4)
* [v5 2/6] [StatusBarFooter: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/174eea25410ced9ba86f7caa2fe257567b4f774c)
* [v5 3/6] [ResultDisplay: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/32611fe21f6d8f05ae8173758029d60928a5ab6e)
* [v5 4/6] [PersonListPanel: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/542941ed58ce772bc9dba08d57ff996c952666d4)
* [v5 5/6] [BrowserPanel: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/b2c0d4a5d2cf61bdeb2ac7515a9fe4817d55fe1f)
* [v5 6/6] [MainWindow: Refactor out component creation logic](https://github.com/se-edu/addressbook-level4/pull/497/commits/66416826e00307f849761f5db5c6bc6ef4303d9d)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/5cc85fcf11b40d1660d0b6be20a3ef90bc71c1c1...66416826e00307f849761f5db5c6bc6ef4303d9d)) ([:chart_with_upwards_trend: Interdiff between v4 and v5](https://github.com/se-edu/addressbook-level4/compare/0f950de20d71862e6af8283b34046d97646e1814~1...0f950de20d71862e6af8283b34046d97646e1814))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/497/5/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

**Comment by yamgent** [comment]
@MightyCupcakes Nice spot, thanks!

**Comment by CanIHasReview** [comment]
# v6

@yamgent submitted v6 for review.

* [v6 1/6] [CommandBox: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/854e28b4a9b9ee280c77d59b576ec1bc86b19f8e)
* [v6 2/6] [StatusBarFooter: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/1dfbcb03e4a8f89c9f766522bc9abcd763c9a0fa)
* [v6 3/6] [ResultDisplay: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/9f9c7ec2c765ec63168a35dc4db8eb2ee73ddf2a)
* [v6 4/6] [PersonListPanel: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/ae467dfe99158aab0db6406e98279dacb7ef5cd0)
* [v6 5/6] [BrowserPanel: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/7641771fdd855b738d0fac85c7f9e61647f0384d)
* [v6 6/6] [MainWindow: Refactor out component creation logic](https://github.com/se-edu/addressbook-level4/pull/497/commits/00ba71ebd86268e4ed2084a0832e88fff746b670)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/5cc85fcf11b40d1660d0b6be20a3ef90bc71c1c1...00ba71ebd86268e4ed2084a0832e88fff746b670)) ([:chart_with_upwards_trend: Interdiff between v5 and v6](https://github.com/se-edu/addressbook-level4/compare/e368b1984657d8cacaed10aed8bf68fce3046f75~1...e368b1984657d8cacaed10aed8bf68fce3046f75))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/497/6/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 3]
```diff
@@ -119,7 +120,10 @@ void fillInnerParts() {
...
         new ResultDisplay(getResultDisplayPlaceholder());
         new StatusBarFooter(getStatusbarPlaceholder(), prefs.getAddressBookFilePath());
-        new CommandBox(getCommandBoxPlaceholder(), logic);
+        CommandBox commandBox = new CommandBox(logic);
+
+        SplitPane.setResizableWithParent(commandBoxPlaceholder, false);
+        commandBoxPlaceholder.getChildren().add(commandBox.getRoot());
     }
 
     private StackPane getCommandBoxPlaceholder() {
```
Comment (Zhiyuan-Amos) [inline src/main/java/seedu/address/ui/MainWindow.java]: this method will become unused if I'm not wrong. Same for other commits getter methods.

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 4]
```diff
@@ -115,11 +117,60 @@ private void setAccelerator(MenuItem menuItem, KeyCombination keyCombination) {
...
+        statusbarPlaceholder.getChildren().add(statusBarFooter.getRoot());
+    }
+
+    /**
+     * Creates a new {@code CommandBox} and places it inside its placeholder.
+     */
+    private void setupCommandBox() {
+        CommandBox commandBox = new CommandBox(logic);
+
+        SplitPane.setResizableWithParent(commandBoxPlaceholder, false);
```
Comment (Zhiyuan-Amos) [inline src/main/java/seedu/address/ui/MainWindow.java]: not in this PR, but just wondering why are we calling `SplitPane.setResizable...` when commandBoxPlaceholder is a StackPane, not a SplitPane.

**Comment by CanIHasReview** [comment]
# v7

@yamgent submitted v7 for review.

* [v7 1/7] [CommandBox: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/93c52a77f1882fee03a40bbccc76fee61d8064f6)
* [v7 2/7] [StatusBarFooter: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/0e033328210a3af74a4fcd25ed3943f3a2b07600)
* [v7 3/7] [ResultDisplay: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/52cc7f8223723324538dda18700c683bcfa48a53)
* [v7 4/7] [PersonListPanel: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/540fb202fc6d42ca73eced4df49498055f8dc850)
* [v7 5/7] [BrowserPanel: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/88dd9579406abc8ddecf54ddccc109d0c0411df0)
* [v7 6/7] [MainWindow: Refactor out component creation logic](https://github.com/se-edu/addressbook-level4/pull/497/commits/1882ea974f920fff8bbe3346ca95868d04eb3036)
* [v7 7/7] [MainWindow: Remove getXPlaceholder\(\) methods](https://github.com/se-edu/addressbook-level4/pull/497/commits/7f872a9646b66a692ffe6a79acd77017ad727333)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/425582ae92d17a02ed45464fb87e52d3e0034b0c...7f872a9646b66a692ffe6a79acd77017ad727333)) ([:chart_with_upwards_trend: Interdiff between v6 and v7](https://github.com/se-edu/addressbook-level4/compare/9dc23428448c83575a2c7fe185fba3cef00177b0~1...9dc23428448c83575a2c7fe185fba3cef00177b0))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/497/7/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 5]
```diff
@@ -115,27 +117,60 @@ private void setAccelerator(MenuItem menuItem, KeyCombination keyCombination) {
...
-        return commandBoxPlaceholder;
+    /**
+     * Creates a new {@code BrowserPanel} and places it inside its placeholder.
+     */
+    private void setupBrowserPanel() {
+        browserPanel = new BrowserPanel();
+
+        browserPlaceholder.getChildren().add(browserPanel.getRoot());
+        // To prevent triggering events for typing inside the loaded Web page.
+        browserPlaceholder.setOnKeyPressed(Event::consume);
```
Comment (pyokagan) [inline src/main/java/seedu/address/ui/MainWindow.java]: MainWindow shouldn't need to be aware of this? Can this code be moved into `BrowserPanel` instead?

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 6]
```diff
@@ -115,27 +117,60 @@ private void setAccelerator(MenuItem menuItem, KeyCombination keyCombination) {
...
 
-    private StackPane getStatusbarPlaceholder() {
-        return statusbarPlaceholder;
+    /**
+     * Creates a new {@code PersonListPanel} and places it inside its placeholder.
+     */
+    private void setupPersonListPanel() {
+        personListPanel = new PersonListPanel(logic.getFilteredPersonList());
+
+        SplitPane.setResizableWithParent(personListPanelPlaceholder, false);
```
Comment (pyokagan) [inline src/main/java/seedu/address/ui/MainWindow.java]: afaik this can be set in FXML?

**Comment by CanIHasReview** [comment]
# v8

@yamgent submitted v8 for review.

* [v8 1/8] [CommandBox: Remove call to SplitPane\#setResizableWithParent\(\)](https://github.com/se-edu/addressbook-level4/pull/497/commits/f7a5766af3d09475af6f9b276c8c26dc624b502d)
* [v8 2/8] [CommandBox: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/4acbb6cd76c5a683c70ff519eeecea775d2ab2df)
* [v8 3/8] [StatusBarFooter: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/dac78051e25237a925636246c6f873e2e9bebd65)
* [v8 4/8] [ResultDisplay: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/baf00d32843e028a9875e93f413f818e8e5346f8)
* [v8 5/8] [PersonListPanel: Set SplitPane\.resizableWithParent in \.fxml](https://github.com/se-edu/addressbook-level4/pull/497/commits/0ec45fdce6221abfbc5d1f2f59f23f70c245cb15)
* [v8 6/8] [PersonListPanel: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/d222be9a755cb75a82d571e56940eb7a79b307a4)
* [v8 7/8] [BrowserPanel: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/4810e4a6fa923f427260ee94eb353df832b05d14)
* [v8 8/8] [MainWindow: Remove getXPlaceholder\(\) methods](https://github.com/se-edu/addressbook-level4/pull/497/commits/5736ee210f88fa8ce65c0bd14f71bc91612f9d35)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/48775e45f9e37d5c00b99f7241d2856b3e2eac9e...5736ee210f88fa8ce65c0bd14f71bc91612f9d35)) ([:chart_with_upwards_trend: Interdiff between v7 and v8](https://github.com/se-edu/addressbook-level4/compare/a5afbc3ce0632ae91776130c4d5d767510a12afd~1...a5afbc3ce0632ae91776130c4d5d767510a12afd))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/497/8/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

**Comment by yamgent** [comment]
# v8

* [v8 1/8] and [v8 5/8] are new commits.
   * For [v8 1/8], the attribute is completely removed because CommandBox's placeholder isn't even inside a SplitPane, so that call is erroneous.
* Paul's requested changes have been made

File: src/main/resources/view/MainWindow.fxml [hunk 7]
```diff
@@ -42,7 +42,7 @@
       <padding>
         <Insets top="10" right="10" bottom="10" left="10" />
       </padding>
-      <StackPane fx:id="personListPanelPlaceholder" VBox.vgrow="ALWAYS" />
+      <StackPane fx:id="personListPanelPlaceholder" VBox.vgrow="ALWAYS" SplitPane.resizableWithParent="false" />
```
Comment (Zhiyuan-Amos) [inline src/main/resources/view/MainWindow.fxml]: @pyokagan Wang Leng and I tried making this attribute to both `true` and `false`, and we noticed no difference for the `PersonListPanel`.

When we put this line of code in Vbox instead (the direct child node of the SplitPane), it works. As such, @yamgent will make another commit to shift it to the Vbox instead.

**Comment by CanIHasReview** [comment]
# v9

@yamgent submitted v9 for review.

* [v9 1/9] [CommandBox: Remove call to SplitPane\#setResizableWithParent\(\)](https://github.com/se-edu/addressbook-level4/pull/497/commits/f7a5766af3d09475af6f9b276c8c26dc624b502d)
* [v9 2/9] [CommandBox: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/4acbb6cd76c5a683c70ff519eeecea775d2ab2df)
* [v9 3/9] [StatusBarFooter: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/dac78051e25237a925636246c6f873e2e9bebd65)
* [v9 4/9] [ResultDisplay: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/baf00d32843e028a9875e93f413f818e8e5346f8)
* [v9 5/9] [PersonListPanel: Set SplitPane\.resizableWithParent in \.fxml](https://github.com/se-edu/addressbook-level4/pull/497/commits/d1bd1cd1a49ab1fe3159cf955aa37de686db3cf5)
* [v9 6/9] [PersonListPanel: Fix SplitPane\.resizableWithParent not working](https://github.com/se-edu/addressbook-level4/pull/497/commits/7608a9c595d4779a9d3a56f70d8f6d6500d6b919)
* [v9 7/9] [PersonListPanel: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/9b45ac30273c71d112a1b34bf2b55732868bf561)
* [v9 8/9] [BrowserPanel: Remove placeholder from constructor](https://github.com/se-edu/addressbook-level4/pull/497/commits/1d780e1567479dce3bfdbd15891f44b70f5ab7fb)
* [v9 9/9] [MainWindow: Remove getXPlaceholder\(\) methods](https://github.com/se-edu/addressbook-level4/pull/497/commits/d511d69e7f4ee1a07d83c8a6d0d0d1f53eeb391d)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/48775e45f9e37d5c00b99f7241d2856b3e2eac9e...d511d69e7f4ee1a07d83c8a6d0d0d1f53eeb391d)) ([:chart_with_upwards_trend: Interdiff between v8 and v9](https://github.com/se-edu/addressbook-level4/compare/7031fb8c41ad4a0baca4c0460f73791be57ac6d8~1...7031fb8c41ad4a0baca4c0460f73791be57ac6d8))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/497/9/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

File: src/main/resources/view/BrowserPanel.fxml [hunk 8]
```diff
@@ -4,5 +4,6 @@
 <?import javafx.scene.web.WebView?>
 
 <AnchorPane xmlns:fx="http://javafx.com/fxml/1">
-  <WebView fx:id="browser" />
+  <WebView fx:id="browser"
+    AnchorPane.topAnchor="0.0" AnchorPane.bottomAnchor="0.0" AnchorPane.leftAnchor="0.0" AnchorPane.rightAnchor="0.0"/>
```
Comment (pyokagan) [inline src/main/resources/view/BrowserPanel.fxml]: Maybe add some cleanup commits on top to convert these AnchorPanes to StackPanes?
